File: php/switchcase.php
<?php

// Fungsi untuk meminta nama sampai valid
function inputNama() {
    while (true) {
        echo "Masukkan nama: ";
        $nama = trim(fgets(STDIN));

        if ($nama == "") {
            echo "Nama tidak boleh kosong! Coba lagi.\n";
            continue;
        }

        echo "Apakah nama sudah benar? (true/false): ";
        $valid = trim(fgets(STDIN));

        if (strtolower($valid) === "true") {
            return $nama;
        }

        echo "Nama salah, masukkan kembali.\n\n";
    }
}

// Input nama
$nama = inputNama();

// Input NIM
echo "\nMasukkan NIM (angka saja): ";
$nim = trim(fgets(STDIN));

$saldo = number_format($nim, 0, ',', '.');

echo "\n===== DATA AWAL =====\n";
echo "Nama  : $nama\n";
echo "Saldo : Rp $saldo\n";

while (true) {
    echo "\n===== MENU ATM =====\n";
    echo "1. Cek Saldo\n";
    echo "2. Tarik Tunai\n";
    echo "3. Setor Tunai\n";
    echo "4. Transfer\n";
    echo "5. Keluar\n";
    echo "Pilih menu: ";
    $menu = trim(fgets(STDIN));

    switch ($menu) {
        case 1:
            echo "Saldo anda: Rp $saldo\n";
            break;

        case 2:
            echo "Masukkan nominal tarik: ";
            $tarik = trim(fgets(STDIN));
            echo "Anda menarik Rp " . number_format($tarik, 0, ',', '.') . "\n";
            break;

        case 3:
            echo "Masukkan nominal setor: ";
            $setor = trim(fgets(STDIN));
            echo "Anda menyetor Rp " . number_format($setor, 0, ',', '.') . "\n";
            break;

        case 4:
            echo "Masukkan nominal transfer: ";
            $transfer = trim(fgets(STDIN));
            echo "Anda transfer Rp " . number_format($transfer, 0, ',', '.') . "\n";
            break;

        case 5:
            exit("Terima kasih telah menggunakan ATM!\n");

        default:
            echo "Menu tidak tersedia.\n";
    }
}

File: c atau c++/switchcase.cpp
#include <iostream>
#include <iomanip>
#include <string>
using namespace std;

string inputNama() {
    string nama;
    string valid;

    while (true) {
        cout << "Masukkan Nama Anda: ";
        getline(cin, nama);

        if (nama.empty()) {
            cout << "Harap Nama Tidak Boleh Kosong!\n";
            continue;
        }

        cout << "Apakah Nama Anda Sudah Benar? (true/false): ";
        cin >> valid;
        cin.ignore();

        if (valid == "true" || valid == "TRUE") {
            return nama;
        }

        cout << "Nama Salah, Harap Masukkan Kembali.\n\n";
    }
}

int main() {
    // Input nama
    string nama = inputNama();

    // Input NIM
    long long nim;
    cout << "\nMasukkan NIM Anda(angka saja): ";
    cin >> nim;
    cin.ignore();

    long long saldo = nim;

    cout << "\n===== DATA AWAL =====\n";
    cout << "Nama  : " << nama << endl;
    cout << "Saldo : Rp " << saldo << endl;

    int menu;
    while (true) {
        cout << "\n===== MENU ATM =====\n";
        cout << "1. Cek Saldo\n";
        cout << "2. Tarik Tunai\n";
        cout << "3. Setor Tunai\n";
        cout << "4. Transfer\n";
        cout << "5. Keluar\n";
        cout << "Pilih menu: ";
        cin >> menu;

        switch (menu) {
            case 1:
                cout << "Saldo anda: Rp " << saldo << endl;
                break;

            case 2: {
                long long tarik;
                cout << "Nominal tarik: ";
                cin >> tarik;
                cout << "Anda menarik Rp " << tarik << endl;
                break;
            }

            case 3: {
                long long setor;
                cout << "Nominal setor: ";
                cin >> setor;
                cout << "Anda menyetor Rp " << setor << endl;
                break;
            }

            case 4: {
                long long transfer;
                cout << "Nominal transfer: ";
                cin >> transfer;
                cout << "Anda transfer Rp " << transfer << endl;
                break;
            }

            case 5:
                cout << "Terima kasih telah menggunakan ATM!\n";
                return 0;

            default:
                cout << "Menu tidak valid!\n";
        }
    }
}
